homeassistant/components/smtp/config_flow.py

```diff
@@ -0,0 +1,197 @@
+"""Config flow for SMTP integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import (
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PLATFORM,
+    CONF_PORT,
+    CONF_RECIPIENT,
+    CONF_SENDER,
+    CONF_TIMEOUT,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import DOMAIN as HOMEASSISTANT_DOMAIN
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+from homeassistant.helpers.selector import (
+    BooleanSelector,
+    BooleanSelectorConfig,
+    NumberSelector,
+    NumberSelectorConfig,
+    NumberSelectorMode,
+    SelectSelector,
+    SelectSelectorConfig,
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import (
+    CONF_DEBUG,
+    CONF_ENCRYPTION,
+    CONF_SENDER_NAME,
+    CONF_SERVER,
+    DEFAULT_DEBUG,
+    DEFAULT_ENCRYPTION,
+    DEFAULT_HOST,
+    DEFAULT_PORT,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+    ENCRYPTION_OPTIONS,
+)
+from .notify import MailNotificationService
+
+_LOGGER = logging.getLogger(__name__)
+
+
+BASE_DATA_FIELDS = {
+    vol.Optional(CONF_NAME): TextSelector(
+        TextSelectorConfig(type=TextSelectorType.TEXT)
+    ),
```

Comment:
Hm, we normally don't want to collect names in config flows, but I'm also lacking a better idea since we are also collecting multiple recipients.

Otherwise this looks good to me, I've tested this and it works like a charm, great work, thanks 👍

Reply:
The name seems a key feature to determine the notifier service name, I left it away at first, but then that will break the existing configurations.

Reply:
Okay, thank I think this is okay